partner_contact_sale_info_propagation/models/res_partner.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+from lxml import etree
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.multi
+    def write(self, vals):
+        """Propagate Salesperson and Sales Channel change in the partner to the
+        child contacts."""
+        for record in self:
+            if vals.get('user_id'):
+                childs = record.mapped('child_ids').filtered(
+                    lambda r: not r.user_id or r.user_id == record.user_id)
+                if childs:
+                    childs.write({'user_id': vals['user_id']})
+            if vals.get('team_id'):
+                childs = record.mapped('child_ids').filtered(
+                    lambda r: not r.team_id or r.team_id == record.team_id)
+                if childs:
+                    childs.write({'team_id': vals['team_id']})
+        return super(ResPartner, self).write(vals)
+
+    @api.model
+    def create(self, vals):
+        if vals.get('parent_id'):
+            if not vals.get('user_id'):
```

Comment:
@ernestotejeda IMO is  better
```
 if not vals.get('user_id') and not vals.get('team_id'):
    vals.update({
        'user_id': self.browse(vals['parent_id']).user_id.id,
        'team_id': self.browse(vals['parent_id']).team_id.id,
    })
```

Reply:
This should be:
```python
if 'user_id' not in vals and 'team_id' not in vals:
```
or empty fields won't be propagated.

Reply:
But any way, it's correct to split the `if`, as there are 2 conditions inside.